### PR TITLE
fix antag bugs

### LINKED ISSUE #27306: Traitor gamemode starts with 1 player
## Description
If I run a local server on Release build, and play there alone, Traitor gamemode is selected even though only I am on the server.

**Reproduction**
Build the game on Release, `dotnet build --configuration Release`, run it, just try to play a normal shift.
For me the game often tried to pick Nukies, decides not to because 1 players is less than 35, then goes for Traitor gamemode, which it succeeds.

Also the syndicate agent objectives I got probably aren't what I should've gotten.

**Screenshots**
20240425_05h03m29s_grim

**Additional context**
Just a guess, but it might have something to do with recently merged #25786

### LINKED ISSUE #27312: Antag objectives are weird
## Description
People are getting tons of objectives and antag safe roles are getting antag

And too many objectives for the difficulty 

**Reproduction**
Become syndie 

image

image

Research assistant syndie and ensure fellow TRAITOR secoff objective 

**Additional context**

### LINKED ISSUE #27313: Admin visibility of antags is inconsistent.
The status of Syndicate traitors appears to be inconsistently viewable by admins currently. Whilst some were visible in the ahelp menu, some were not, despite actually being antagonists. Additionally, chat notifications for players causing explosions that should normally note when antagonists are responsible do not.

Unsure how to reproduce or what the cause of the issue is as sometimes the traitors are visible to admins, sometimes they are not.

image_2024-04-25_163523706

This person was an antagonist and should have been marked as such in this message during the detonation.

### PR DESCRIPTION
**Pull request by EmoGarbage404**
## About the PR

Main Fixes
- fixes issue where latejoins weren't actually checked to see if they were valid to be antags (sec/command antags)
- fixes fucked variable in warops code that lead to TC not getting distributed
- changed how mind components were applied to actually update admin antag panel
- fixed traitors having like 7 morbillion objectives.
- fixed delayed gamerules starting even if they don't have enough players

Fixes #27313 
Fixes #27312 
Fixes #27306 

## Media

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl:
- fix: Traitors now have the correct number of objectives.
- fix: Declaring war now gives TC again.
- fix: Latejoining antagonists will now properly exclude command and security staff.
ADMIN:
- fix: Fixed antag status being broken in overlay and UI.

### CODE REVIEW

File: Content.Server/GameTicking/Rules/Components/NukeopsRuleComponent.cs [hunk 1]
```diff
@@ -75,7 +75,7 @@ public sealed partial class NukeopsRuleComponent : Component
     ///     Minimal operatives count for war declaration
     /// </summary>
     [DataField]
-    public int WarDeclarationMinOps = 4;
+    public int WarDeclarationMinOps = 1;//4;
```
Comment (ShadowCommander) [inline Content.Server/GameTicking/Rules/Components/NukeopsRuleComponent.cs:78]: Is this for testing?

File: Content.Server/GameTicking/Rules/Components/TraitorRuleComponent.cs [hunk 2]
```diff
@@ -71,5 +71,5 @@ public enum SelectionState
     public int StartingBalance = 20;
 
     [DataField]
-    public int MaxDifficulty = 20;
+    public int MaxDifficulty = 5;
```
Comment (ShadowCommander) [inline Content.Server/GameTicking/Rules/Components/TraitorRuleComponent.cs:74]: Is this for testing?

Comment (EmoGarbage404) [reply Content.Server/GameTicking/Rules/Components/TraitorRuleComponent.cs:74]: This isnt

**Review by ShadowCommander — APPROVED** [review]

File: Content.Server/Antag/AntagSelectionSystem.cs [hunk 3]
```diff
@@ -310,42 +324,45 @@ public void MakeAntag(Entity<AntagSelectionComponent> ent, ICommonSession? sessi
...
     public AntagSelectionPlayerPool GetPlayerPool(Entity<AntagSelectionComponent> ent, List<ICommonSession> sessions, AntagSelectionDefinition def)
     {
-        var primaryList = new List<ICommonSession>();
-        var secondaryList = new List<ICommonSession>();
-        var fallbackList = new List<ICommonSession>();
-        var rawList = new List<ICommonSession>();
+        var preferredList = new List<ICommonSession>();
+        var secondBestList = new List<ICommonSession>();
+        var unwantedList = new List<ICommonSession>();
+        var invalidList = new List<ICommonSession>();
```
Comment (deltanedas) [inline Content.Server/Antag/AntagSelectionSystem.cs:330]: will this still not pick from unwanted and invalid if the others fail

**Pull request merged**